eZ/Bundle/EzPublishCoreBundle/DependencyInjection/Compiler/UrlWildcardRouterPass.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/**
+ * File containing the ChainRoutingPass class.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Bundle\EzPublishCoreBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\DependencyInjection\Reference;
+
+/**
+ * The UrlWildcardRouterPass registers UrlWildcardRouter if enabled in the configuration.
+ * It needs to be executed before \eZ\Bundle\EzPublishCoreBundle\DependencyInjection\Compiler\ChainRoutingPass.
+ */
+class UrlWildcardRouterPass implements CompilerPassInterface
+{
+    public function process(ContainerBuilder $container): void
+    {
+        if (
+            !$container->hasParameter('ezpublish.url_wildcards.enabled') ||
+            !$container->hasParameter('ezpublish.urlwildcard_router.class')
+        ) {
+            return;
+        }
+
+        if ($container->getParameter('ezpublish.url_wildcards.enabled') === true) {
```

Comment:
TBH I have strong reservation about this entire compiler pass. I don't really see a point in parametrizing container building based on some settings. It's difficult to maintain as our service definition exist in Yaml files.
Why do we even need to make it optional?

Reply:
Performance overhead  for those not using it (extra lookup for every request) and there is no UI to deal with this in 2.5 _(yet)_ so it's more for migration feature parity for now.

Reply:
@alongosz, as @andrerom explained, it's meant to be an _optional_ feature just for the customers who migrated from eZ Publish and still have to use UrlWildcards defined in the past (for SEO / migration reasons for instance). It was discussed with PMs and @lserwatka.